### PR TITLE
feat(Planet Minecraft): add presence

### PR DESCRIPTION
**Pull request by TheCardinalSystem**
## Description 
Planet Minecraft is a family friendly community that shares and respects the creative works and interests of others. They have a variety of entertaining ways to explore and upload content, socialize and find like minded creative members from around the world that love the game of Minecraft.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/P/Planet Minecraft/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		"id": "283380624687890433",
+		"name": "cardinalsystem"
+	},
+	"service": "Planet Minecraft",
+	"description": {
+		"en": "Planet Minecraft is a family friendly community that shares and respects the creative works and interests of others. They have a variety of entertaining ways to explore and upload content, socialize and find like minded creative members from around the world that love the game of Minecraft."
+	},
+	"url": "www.planetminecraft.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/YcaaiHk.png",
```
Comment (theusaf) [inline websites/P/Planet Minecraft/metadata.json:13]: The [logo](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo) should be 512x512px.

File: websites/P/Planet Minecraft/presence.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+	if (!strings) strings = await getStrings();
+
+	const pathname = document.location.pathname.replaceAll(/\/$/g, "");
+
+	if (pathname === "") {
+		presenceData.state = strings.viewHome;
+		delete presenceData.details;
+	} else {
+		const url = `${location.protocol}//${location.host}${location.pathname}`;
```
Comment (theusaf) [inline websites/P/Planet Minecraft/presence.ts:35]: Use `document.location` properties over `window` or global `location` properties.
```ts
const { pathname, protocol, host, href } = document.location;
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/P/Planet Minecraft/presence.ts [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+		if (pathname.startsWith("/member"))
+			memberActivity(presenceData, pathname, url);
+		else if (
+			pathname.startsWith("/account/manage") &&
+			pathname.endsWith("/new")
+		) {
+			presenceData.details = "Managing Submissions";
+			presenceData.state = document.querySelector("#manage_header").textContent;
+		} else {
+			switch (pathname) {
```
Comment (DarkVillager) [inline websites/P/Planet Minecraft/presence.ts:45]: Why not make this one switch (true)

File: websites/P/Planet Minecraft/presence.ts [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+
+	if (!strings) strings = await getStrings();
+
+	const pathname = document.location.pathname.replaceAll(/\/$/g, "");
+
+	if (pathname === "") {
+		presenceData.state = strings.viewHome;
+		delete presenceData.details;
+	} else {
+		const url = `${document.location.protocol}//${document.location.host}${document.location.pathname}`;
```
Comment (DarkVillager) [inline websites/P/Planet Minecraft/presence.ts:35]: Isnt href the exact same?

Comment (TheCardinalSystem) [reply websites/P/Planet Minecraft/presence.ts:35]: `href` includes query params which could contain sensitive data.

File: websites/P/Planet Minecraft/presence.ts (hunk 3 above)
Comment (TheCardinalSystem) [reply websites/P/Planet Minecraft/presence.ts:45]: This is a very interested switch usage; I've never seen it before. I would think that my current approach performs better than the one you suggested, but this is going off a limited knowledge of JS under the hood. I am interested to see what [other people have to say](https://stackoverflow.com/q/77787154/5645656).

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**